test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenUpdateSpecs.java

```diff
@@ -587,6 +593,75 @@ public HapiApiSpec updateHappyPath() {
 				);
 	}
 
+	public HapiApiSpec updateTokenTreasuryHappyPath() {
+		return defaultHapiSpec("UpdateTokenTreasuryHappyPath")
+				.given(
+						cryptoCreate(TOKEN_TREASURY),
+						cryptoCreate("newTokenTreasury"),
+						newKeyNamed("adminKey"),
+						newKeyNamed("supplyKey"),
+						tokenCreate("primary")
+								.treasury(TOKEN_TREASURY)
+								.initialSupply(500)
+								.adminKey("adminKey")
+								.supplyKey("supplyKey")
+				)
+				.when(
+						tokenAssociate("newTokenTreasury", "primary"),
+						tokenUpdate("primary")
+								.treasury("newTokenTreasury").via("tokenUpdateTxn")
+				)
+				.then(
+						getAccountBalance(TOKEN_TREASURY)
+								.hasTokenBalance("primary", 0),
+						getAccountBalance("newTokenTreasury")
+								.hasTokenBalance("primary", 500),
+						getTokenInfo("primary")
+								.hasTreasury("newTokenTreasury")
+								.hasTotalSupply(500),
+						getTxnRecord("tokenUpdateTxn").logged()
+				);
+	}
+
+	public HapiApiSpec updateNftTreasuryHappyPath() {
```

Comment:
I think that a good test is verifying that if the new treasury has NFT balance, the `TokenUpdate` is not performed and the response code is `TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES`